packages/graph/hash_graph/lib/graph/src/types/schema/data.rs

```diff
@@ -0,0 +1,316 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+use crate::types::schema::{Uri, Validate, ValidationError};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(
+    try_from = "DataTypeRepr",
+    tag = "kind",
+    rename = "dataType",
+    rename_all = "camelCase"
+)]
+pub struct DataType {
+    #[serde(rename = "$id")]
+    id: Uri,
+    title: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    description: Option<String>,
+    #[serde(rename = "type")]
+    ty: String,
+    #[serde(flatten)]
+    additional_properties: HashMap<String, serde_json::Value>,
+}
+
+impl DataType {
+    /// Creates a new `PropertyType` without validating.
+    #[must_use]
+    pub fn new_unchecked(
+        id: Uri,
+        title: impl Into<String>,
+        description: impl Into<Option<String>>,
+        ty: impl Into<String>,
+        additional_properties: HashMap<String, serde_json::Value>,
+    ) -> Self {
+        Self {
+            id,
+            title: title.into(),
+            description: description.into(),
+            ty: ty.into(),
+            additional_properties,
+        }
+    }
+
+    /// Creates a new `PropertyType`.
+    ///
+    /// # Errors
+    ///
+    /// - [`ValidationError`] if validation is failing.
+    pub fn new(
+        id: Uri,
+        title: impl Into<String>,
+        description: impl Into<Option<String>>,
+        ty: impl Into<String>,
+        additional_properties: HashMap<String, serde_json::Value>,
+    ) -> Result<Self, ValidationError> {
+        let property_type = Self::new_unchecked(id, title, description, ty, additional_properties);
```

Comment:
data_type?

Reply:
Fixed in 3d717d5b2a8e8c786869afce31eca9fe95f4e21e